contracts/teller-network/Escrow.sol

```diff
@@ -61,43 +70,113 @@ contract Escrow is Pausable, MessageSigned {
      */
-    function create(address payable _buyer, uint _amount, address _token, uint _expirationTime) public payable whenNotPaused {
+    function fund(uint _escrowId, uint _tokenAmount, uint _expirationTime) public payable whenNotPaused {
+        require(_escrowId < transactions.length, INVALID_ESCROW_ID);
         require(_expirationTime > (block.timestamp + 600), "Expiration time must be at least 10min in the future");
         require(license.isLicenseOwner(msg.sender), "Must be a valid seller to create escrow transactions");
 
-        if(_token == address(0)){
-            require(msg.value == _amount, "ETH amount is required");
+        EscrowTransaction storage trx = transactions[_escrowId];
+
+        if(trx.token == address(0)){
+            require(msg.value == _tokenAmount, "ETH amount is required");
         } else {
             require(msg.value == 0, "Cannot send ETH with token address different from 0");
-            ERC20Token token = ERC20Token(_token);
-            require(token.allowance(msg.sender, address(this)) >= _amount, "Allowance not set for this contract for specified amount");
-            require(token.transferFrom(msg.sender, address(this), _amount), "Unsuccessful token transfer");
+            ERC20Token token = ERC20Token(trx.token);
+            require(token.allowance(msg.sender, address(this)) >= _tokenAmount, "Allowance not set for this contract for specified amount");
+            require(token.transferFrom(msg.sender, address(this), _tokenAmount), "Unsuccessful token transfer");
         }
 
+        transactions[_escrowId].tokenAmount = _tokenAmount;
+        transactions[_escrowId].expirationTime = _expirationTime;
+        transactions[_escrowId].status = EscrowStatus.FUNDED;
+
+        emit Funded(_escrowId, _expirationTime, _tokenAmount);
+    }
+
+
+    /**
+     * @notice Create and fund escrow
+     * @param _buyer Buyer address
+     * @param _tradeAmount Amount buyer is willing to trade
+     * @param _tradeType Indicates if the amount is in crypto or fiat
+     * @param _token Token address. Must be 0 for ETH
+     * @param _tokenAmount How much ether/tokens will be put in escrow
+     * @param _expirationTime Unix timestamp before the transaction is considered expired
+     * @dev Requires contract to be unpaused.
+     *         The seller needs to be licensed.
+     *         The expiration time must be at least 10min in the future
+     *         For eth transfer, _amount must be equals to msg.value, for token transfer, requires an allowance and transfer valid for _amount
+     */
+    function create_and_fund(address payable _buyer, address _token, uint _tokenAmount, uint _expirationTime, uint _tradeAmount, uint8 _tradeType) public payable whenNotPaused {
+        require(license.isLicenseOwner(msg.sender), "Must be a valid seller to create and fund escrow transactions");
```

Comment:
Should this function just called the 2 underlying function?
That will ensure validation are done once and correctly no?